### PR TITLE
Increase new fed activation age

### PR DESCRIPTION
**Pull request by nathanieliov**

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java [hunk 1]
```diff
@@ -88,7 +91,7 @@ void whenOldAndNewFederationArePresentReturnOldFederationByActivationAge() {
         when(provider.getNewFederation()).thenReturn(newFederation);
         when(provider.getOldFederation()).thenReturn(oldFederation);
         when(executionBlock.getNumber()).thenReturn(80L);
-        when(bridgeConstants.getFederationActivationAge()).thenReturn(10L);
+        when(bridgeConstants.getFederationActivationAge(activations)).thenReturn(10L);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java:94]: Should we test this pre and post fork? Same with the test below

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java [hunk 2]
```diff
@@ -184,10 +184,10 @@ private void test_processFundsMigration(
 
         long federationCreationBlockNumber = 5L;
         long federationInMigrationAgeHeight = federationCreationBlockNumber +
-            bridgeConstants.getFederationActivationAge() +
+            bridgeConstants.getFederationActivationAge(activations) +
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java:187]: Looks like we are missing tests in this class also, we need to check that the federation is activated later after fingerroot

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 3]
```diff
@@ -249,7 +249,7 @@ void logCommitFederation() {
         String oldFederationBtcAddress = oldFederation.getAddress().toBase58();
         byte[] newFederationFlatPubKeys = flatKeysAsByteArray(newFederation.getBtcPublicKeys());
         String newFederationBtcAddress = newFederation.getAddress().toBase58();
-        long newFedActivationBlockNumber = executionBlock.getNumber() + CONSTANTS.getFederationActivationAge();
+        long newFedActivationBlockNumber = executionBlock.getNumber() + CONSTANTS.getFederationActivationAge(activations);
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:252]: I'm not opposed to writing a new test, but it will be a kind of integration test(like the tests written for the powpeg). We already have specific unit tests for this method, testing exclusively it. See: https://github.com/rsksmart/rskj/blob/increase-new-fed-activation-age/rskj-core/src/test/java/co/rsk/config/BridgeConstantsTest.java#L69-L105

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java [hunk 4]
```diff
@@ -269,7 +269,7 @@ void testLogCommitFederationBeforeRskip146() {
         }
 
         // Assert new federation activation block number
-        Assertions.assertEquals(15L + constantsMock.getFederationActivationAge(), Long.valueOf(new String(dataList.get(2).getRLPData(), StandardCharsets.UTF_8)).longValue());
+        Assertions.assertEquals(15L + constantsMock.getFederationActivationAge(activations), Long.valueOf(new String(dataList.get(2).getRLPData(), StandardCharsets.UTF_8)).longValue());
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java:272]: Same answer.

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 5]
```diff
@@ -277,7 +277,7 @@ private void testChangePowpeg(
 
         // Move the required blocks ahead for the new powpeg to become active
         // (overriding block number to ensure we don't move beyond the activation phase)
-        blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge();
+        blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge(activations);
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:280]: This is being done here: https://github.com/rsksmart/rskj/blob/increase-new-fed-activation-age/rskj-core/src/test/java/co/rsk/config/BridgeConstantsTest.java#L69-L105

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java (hunk 1 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java:94]: See: https://github.com/rsksmart/rskj/blob/increase-new-fed-activation-age/rskj-core/src/test/java/co/rsk/config/BridgeConstantsTest.java#L69-L105

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java (hunk 2 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java:187]: Same here: https://github.com/rsksmart/rskj/blob/increase-new-fed-activation-age/rskj-core/src/test/java/co/rsk/config/BridgeConstantsTest.java#L69-L105

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java (hunk 1 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java:102]: Done

File: rskj-core/src/main/java/co/rsk/config/BridgeTestNetConstants.java [hunk 6]
```diff
@@ -103,6 +103,7 @@ public class BridgeTestNetConstants extends BridgeConstants {
             AddressBasedAuthorizer.MinimumRequiredCalculation.ONE
         );
 
+        federationActivationAgeLegacy = 60L;
         federationActivationAge = 60L;
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/config/BridgeTestNetConstants.java]: Done

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java (hunk 2 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java]: Done

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 7]
```diff
@@ -275,13 +276,54 @@ private void testChangePowpeg(
...
         Block activationBlock = mock(Block.class);
         doReturn(blockNumber).when(activationBlock).getNumber();
 
+        // assuming fed activation age after rskip383 is greater than legacy fed activation age, we can check that new fed
+        // should not be active at the legacy activation age when RSKIP383 is active
+
+        if (activations.isActive(ConsensusRule.RSKIP383)){
+            ActivationConfig.ForBlock activationsBeforeRSKIP383 = mock(ActivationConfig.ForBlock.class);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:289]: Why not simply use hop activations?

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 8]
```diff
@@ -275,13 +276,54 @@ private void testChangePowpeg(
...
         // (overriding block number to ensure we don't move beyond the activation phase)
-        blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge();
+        blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge(activations);
         Block activationBlock = mock(Block.class);
         doReturn(blockNumber).when(activationBlock).getNumber();
 
+        // assuming fed activation age after rskip383 is greater than legacy fed activation age, we can check that new fed
+        // should not be active at the legacy activation age when RSKIP383 is active
+
+        if (activations.isActive(ConsensusRule.RSKIP383)){
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:288]: Is this condition triggered from any of the existing tests?

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 9]
```diff
@@ -25,6 +25,7 @@
 import co.rsk.peg.*;
 import org.bouncycastle.util.encoders.Hex;
 import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:28]: Unused, please check this before submitting the PR

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 10]
```diff
@@ -249,15 +252,118 @@ void logCommitFederation() {
...
+        // assert fed activation has different values before and after RSKIP383 respectively
+        ActivationConfig.ForBlock activationsAfterRSKIP383 = mock(ActivationConfig.ForBlock.class);
+        when(activationsAfterRSKIP383.isActive(ConsensusRule.RSKIP383)).thenReturn(true);
+
+        long hopNewFedActivationBlockNumber = executionBlock.getNumber() + CONSTANTS.getFederationActivationAge(activationsAfterRSKIP383);
+        assertNotEquals(loggedFedActivationBlockNumber, hopNewFedActivationBlockNumber);
+        assertNotEquals(newFedActivationBlockNumber, hopNewFedActivationBlockNumber);
+
+        // assert fed activation value for RSKIP383 is being properly logged
+        assertEquals(loggedFedActivationBlockNumber, newFedActivationBlockNumber);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:282]: Isn't this covered in logCommitFederation_after_RSKIP383?

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java [hunk 11]
```diff
@@ -26,6 +26,7 @@
 import co.rsk.peg.*;
 import org.bouncycastle.util.encoders.Hex;
 import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java:29]: Unused

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java [hunk 12]
```diff
@@ -45,6 +46,8 @@
 
 import static org.junit.jupiter.api.Assertions.assertArrayEquals;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java:50]: Unused

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 13]
```diff
@@ -275,13 +276,54 @@ private void testChangePowpeg(
...
+            ActivationConfig.ForBlock activationsBeforeRSKIP383 = mock(ActivationConfig.ForBlock.class);
+            when(activationsBeforeRSKIP383.isActive(ConsensusRule.RSKIP383)).thenReturn(false);
+
+            long beforeRskip383blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge(activationsBeforeRSKIP383);
+            Assertions.assertTrue(blockNumber > beforeRskip383blockNumber);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:293]: What's the point of this assertion? Asserting that federation activation age is smaller before fingerroot?

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 14]
```diff
@@ -275,13 +276,54 @@ private void testChangePowpeg(
...
         doReturn(blockNumber).when(activationBlock).getNumber();
 
+        // assuming fed activation age after rskip383 is greater than legacy fed activation age, we can check that new fed
+        // should not be active at the legacy activation age when RSKIP383 is active
+
+        if (activations.isActive(ConsensusRule.RSKIP383)){
+            ActivationConfig.ForBlock activationsBeforeRSKIP383 = mock(ActivationConfig.ForBlock.class);
+            when(activationsBeforeRSKIP383.isActive(ConsensusRule.RSKIP383)).thenReturn(false);
+
+            long beforeRskip383blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge(activationsBeforeRSKIP383);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:292]: This is hard to follow. Should the new fed be active at this point? The variable name doesn't help

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 15]
```diff
@@ -275,13 +276,54 @@ private void testChangePowpeg(
...
+            assertNotEquals(oldPowPegAddress, bridgeSupport.getRetiringFederationAddress());
+        } else {
+            ActivationConfig.ForBlock activationsAfterRSKIP383 = mock(ActivationConfig.ForBlock.class);
+            when(activationsAfterRSKIP383.isActive(ConsensusRule.RSKIP383)).thenReturn(true);
+
+            long afterRskip383blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge(activationsAfterRSKIP383);
+
+            // assert fed activation age is different before and after RSKIP383
+            assertNotEquals(blockNumber, afterRskip383blockNumber);
+            Assertions.assertTrue(blockNumber < afterRskip383blockNumber);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:321]: Not sure this assertions make sense here

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 16]
```diff
@@ -275,13 +276,54 @@ private void testChangePowpeg(
...
+                .withActivations(activations)
+                .withBridgeConstants(bridgeConstants)
+                .withBtcBlockStoreFactory(btcBlockStoreFactory)
+                .withPeginInstructionsProvider(new PeginInstructionsProvider())
+                .build();
+
+            // New active powpeg and retiring powpeg
+            assertNotEquals(newPowPegAddress, bridgeSupport.getFederationAddress());
+            assertEquals(oldPowPegAddress, bridgeSupport.getFederationAddress());
+            assertNotEquals(oldPowPegAddress, bridgeSupport.getRetiringFederationAddress());
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:312]: What's the idea here? Proving that the new fed is not yet active?

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java [hunk 17]
```diff
@@ -20,17 +20,26 @@
 import co.rsk.bitcoinj.core.BtcECKey;
 import co.rsk.bitcoinj.core.NetworkParameters;
 import co.rsk.config.BridgeConstants;
+import co.rsk.config.BridgeMainNetConstants;
+import co.rsk.config.BridgeRegTestConstants;
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java]: Done.

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java [hunk 18]
```diff
@@ -88,22 +99,63 @@ void whenOldAndNewFederationArePresentReturnOldFederationByActivationAge() {
...
 
         assertThat(federationSupport.getActiveFederation(), is(oldFederation));
     }
 
-    @Test
-    void whenOldAndNewFederationArePresentReturnNewFederationByActivationAge() {
-        Federation newFederation = getNewFakeFederation(65);
-        Federation oldFederation = getNewFakeFederation(0);
+    private static Stream<Arguments> provideTestArguments() {
+        BridgeConstants bridgeConstants = BridgeTestNetConstants.getInstance();
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java]: Done.

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java [hunk 19]
```diff
@@ -88,22 +99,63 @@ void whenOldAndNewFederationArePresentReturnOldFederationByActivationAge() {
...
+        when(bridgeConstants.getFederationActivationAge(activations)).thenReturn(10L);
 
         assertThat(federationSupport.getActiveFederation(), is(oldFederation));
     }
 
-    @Test
-    void whenOldAndNewFederationArePresentReturnNewFederationByActivationAge() {
-        Federation newFederation = getNewFakeFederation(65);
-        Federation oldFederation = getNewFakeFederation(0);
+    private static Stream<Arguments> provideTestArguments() {
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java]: Done.

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java [hunk 20]
```diff
@@ -88,22 +99,63 @@ void whenOldAndNewFederationArePresentReturnOldFederationByActivationAge() {
...
+        BridgeConstants bridgeConstants = BridgeTestNetConstants.getInstance();
+        ActivationConfig.ForBlock hop = ActivationConfigsForTest.hop400().forBlock(0);
+        ActivationConfig.ForBlock fingerroot = ActivationConfigsForTest.fingerroot500().forBlock(0);
 
+        return Stream.of(
+            Arguments.of(hop, 65, 0, 80L, false),
+            Arguments.of(hop, 65, 0, 65+bridgeConstants.getFederationActivationAge(hop), true),
+            Arguments.of(fingerroot, 65, 0, 80L, false),
+            Arguments.of(fingerroot, 65, 0, 65+bridgeConstants.getFederationActivationAge(fingerroot), true)
+        );
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java]: Done.

File: rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java [hunk 21]
```diff
@@ -88,22 +99,63 @@ void whenOldAndNewFederationArePresentReturnOldFederationByActivationAge() {
...
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("provideTestArguments")
+    void whenOldAndNewFederationArePresentReturnNewFederationByActivationAge(
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java:127]: - It is covered. I will remove the redundant test.
- Method named also needs to be updated.

File: rskj-core/src/test/java/co/rsk/config/BridgeConstantsTest.java [hunk 22]
```diff
@@ -66,4 +65,41 @@ void test_getFundsMigrationAgeSinceActivationEnd_post_RSKIP374(BridgeConstants b
...
+        return Stream.of(
+            Arguments.of(BridgeMainNetConstants.getInstance(), false),
+            Arguments.of(BridgeTestNetConstants.getInstance(), false),
+            Arguments.of(BridgeRegTestConstants.getInstance(), false)
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("generatorFederationActivationAge")
+    void test_getFederationActivationAge(BridgeConstants bridgeConstants, boolean fieldsHasSameValue) {
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/config/BridgeConstantsTest.java]: Refactored.

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java (hunk 11 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java:29]: Seems that there were some issues with my IntelliJ. I had to restart it and invalidate the cache to make it works right again. However, I'll be more cautious with this next time.

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 9 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:28]: Seems that there were some issues with my IntelliJ. I had to restart it and invalidate the cache to make it works right again. However, I'll be more cautious with this next time.

**Comment by nathanieliov** [comment]
pipeline:run

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java (hunk 16 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java]: Refactored

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java (hunk 15 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java]: Refactored

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java (hunk 14 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java]: Refactored

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java (hunk 12 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java]: Removed

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 23]
```diff
@@ -30,6 +30,7 @@
 import org.ethereum.util.ByteUtil;
 import org.ethereum.vm.PrecompiledContracts;
 import org.ethereum.vm.program.InternalTransaction;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java]: Removed

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java [hunk 24]
```diff
@@ -181,13 +194,19 @@ private void test_processFundsMigration(
         BridgeEventLogger bridgeEventLogger = mock(BridgeEventLogger.class);
 
         Federation oldFederation = bridgeConstants.getGenesisFederation();
+        long federationActivationAge = bridgeConstants.getFederationActivationAge(activations);
+
+        long expectedFederationActivationAge = activations.isActive(ConsensusRule.RSKIP383)? 120L: 60L;
+        if (bridgeConstants instanceof BridgeMainNetConstants){
+            expectedFederationActivationAge = activations.isActive(ConsensusRule.RSKIP383)? 40320L: 18500L;
+        }
+        Assertions.assertEquals(expectedFederationActivationAge, federationActivationAge);
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java]: Refactored

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java (hunk 7 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:287]: Because that's the only RSKIP we care for the `getFederationActivationAge` method. I think it's better to keep it specific for assertions.

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java (hunk 8 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:286]: Yes. See: https://github.com/rsksmart/rskj/blob/1dc22ebddb938e21dc95a49c60ddc25def956256/rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java#L1447

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 10 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java]: Done. Redundant tests were removed.

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java (hunk 13 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java]: For the record, we discussed and agreed to keep this assertion for checking legacy fed activation age was increased on RSKIP383.

**Review by marcos-iov — COMMENTED** [review]
Well done, nicely improved.

Just left one very minor comment

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java [hunk 25]
```diff
@@ -231,6 +161,13 @@ private void test_processFundsMigration(
         Assertions.assertEquals(activations.isActive(ConsensusRule.RSKIP146)? 0 : 1, provider.getPegoutsWaitingForConfirmations().getEntriesWithoutHash().size());
         Assertions.assertEquals(activations.isActive(ConsensusRule.RSKIP146) ? 1 : 0, provider.getPegoutsWaitingForConfirmations().getEntriesWithHash().size());
 
+        Assertions.assertTrue(sufficientUTXOsForMigration.isEmpty());
+        if (!inMigrationAge){
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportProcessFundsMigrationTest.java]: Done.

**Review by marcos-iov — APPROVED** [review]
